tools/proto_tool/ProtoLoader.java

```diff
@@ -50,9 +50,17 @@ public static void toProtobin(
     args.add(Paths.get("./tools/protoc.sh").toAbsolutePath().toString());
 
     for (Path proto : protos) {
-      Path resutingFile =
+      Path resultingFile =
           Paths.get(tempDirectory.toAbsolutePath().toString(), proto.getFileName().toString());
 
+      /*
+       * Proto files may import other proto files
+       * In order to support this, on copying them to temporary directory,
+       * we *rewire* the imports to have flat structure
+       * Example: `import common/files/something.proto` becomes `import something.proto`
+       * Current limitation of this approach is not supporting files with same name
+       * by silently overwriting them.
```

Comment:
Can you easily throw an error when we have a collision?

Reply:
@oferb fixed in https://github.com/google/startup-os/pull/574/commits/1ad957b11fed5f79e36d6508d709d1830fdc9bd8